SevenZip/SevenZipCompressor.cs

```diff
@@ -339,25 +311,50 @@ private void SetCompressionProperties()
                         };
 
                         values.Add(pv);
+                    }
+
+                    foreach (var pair in CustomParameters)
+                    {
+                        names.Add(Marshal.StringToBSTR(pair.Key));
+                        var pv = new PropVariant();
+
+                        #region Validate parameters against compression method.
```

Comment:
This is just for ease of use, as certain parameters only apply to certain compression methods.